SpellingBee/create_spelling_data.py

```
@@ -0,0 +1,320 @@
+
+    train_indices = []
+    for i in range(file_len):
+        if i not in most_similar_words_to_test_set and normalize(i, cur_dict, fname) not in test_normalized_tokens:
+            if whole_words_only and not is_whole_word(i, cur_dict, fname):
+                continue
+            train_indices.append(i)
+        if verbose:
+            if normalize(i, cur_dict, fname) in test_normalized_tokens:
+                print("This word normalized version is in test:", end=' |')
+                print(cur_dict[i], end=' | normalize:')
+                print(normalize(i, cur_dict, fname))
+                count_normal_in_test += 1
+            if i not in most_similar_words_to_test_set and normalize(i, cur_dict, fname) in test_normalized_tokens:
+                print("Not 20 closest, but it's normalized version is in test:", end=' |')
+                print(i, end=' |')
+                print(cur_dict[i], end=' | normalize:')
+                print(normalize(i, cur_dict, fname))
+                count_normal_in_test_and_not_similar += 1
+            print("count_normal_in_test:", count_normal_in_test)
+            print("count_normal_in_test_and_not_similar:", count_normal_in_test_and_not_similar)
+
+    return np.random.permutation(train_indices)
+
+
+def get_file_len(fname):
+    with open(fname) as f:
+        for i, l in enumerate(f):
+            pass
+    return i + 1
+
+
+def get_random_split(fname, cur_dict, file_len, test_indices, whole_words_only):
+    train_indices = []
+    if whole_words_only:
+        indices_to_choose_from = [i for i in range(file_len) if is_whole_word(i, cur_dict, fname)]
+    else:
+        indices_to_choose_from = range(file_len)
+
+    for i in indices_to_choose_from:
+        if i not in test_indices:
+            train_indices.append(i)
+    return np.random.permutation(train_indices)
+
+
+def get_split_indicies(fname, cur_dict, file_len, split_state, train_size, whole_words_only, test_indices=None):
```

Why do you always return `None` as the second value from this function?
Also please raise an error if no if passes